rqt_bag/src/rqt_bag/timeline_menu.py

```diff
@@ -45,22 +45,18 @@ def __init__(self, popup_name, timeline, viewer_type, topic):
         self._viewer_type = viewer_type
         self._topic = topic
         self._viewer = None
+        self._is_listening = False
+
 
     def hideEvent(self, event):
-        # check that we were actually hidden. hide events can also happen if a
-        # window is minimized, but isVisible will still be true if a window is
-        # only minimized and not closed
-        if not self.isVisible():
-            # remove the viewer
-            if self._viewer:
-                self._timeline.remove_listener(self._topic, self._viewer)
-                self._viewer.close()
-                self._viewer = None
+        if self._is_listening:
+            self._timeline.remove_listener(self._topic, self._viewer)
+            self._is_listening = False
 
-            # clean out the layout
-            while self.layout().count() > 0:
-                item = self.layout().itemAt(0)
-                self.layout().removeItem(item)
+    def showEvent(self, event):
```

Comment:
According to Qt documentation: 

> void QWidget::hideEvent(QHideEvent * event)
> This event handler can be reimplemented in a subclass to receive widget hide events. The event is passed in the event parameter.
> Hide events are sent to widgets immediately after they have been hidden.

In my understangind, this means that the events are called when the parent class Hide or Show was called (therefore there is no need to call them again). 

This is not true for the show() method, and you can see that it indeed calls the show method of the parent (which BTW leads to a call to showEvent). But I would not add that call into showEvent() and hideEvent()